step/src/main/java/com/google/sps/DatabaseHandler.java

```diff
@@ -15,43 +15,135 @@
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.EntityNotFoundException;
 
 public final class DatabaseHandler {
   
-  // TODO: replace with Datastore
-  private static Map<Long, User> idMap = new HashMap<>();
+  // Accessing Datastore
+  private static DatastoreService datastore =
+         DatastoreServiceFactory.getDatastoreService();
 
   private DatabaseHandler() {
 
   }
 
-  // TODO: call this method when a user first signs up
   // Method for adding a user to the database
-  public static void addUserToDatabase(User user) {
-    idMap.put(user.getId(), user);    
+  public static void addUser(String firstName, String lastName, int dayBirth,
+    int monthBirth, int yearBirth, String email, String id) {
+    Entity entity = new Entity("User", id);
+    entity.setProperty("firstName", firstName);
+    entity.setProperty("lastName", lastName);
+    entity.setProperty("dayBirth", dayBirth);
+    entity.setProperty("monthBirth", monthBirth);
+    entity.setProperty("yearBirth", yearBirth);
+    entity.setProperty("email", email);
+    entity.setProperty("id", id);
+    datastore.put(entity);    
   }
+ 
+  // Adding a user's notificaton to the databse
+  public static void addNotification(String firstId, String secondId,
+      long timestamp, String type) {
+    Entity entity = new Entity("Notification");
+    entity.setProperty("userId", firstId);
+    entity.setProperty("otherUserId", secondId);
+    entity.setProperty("timestamp", timestamp);
+    entity.setProperty("type", type);
+    datastore.put(entity); 
+  }
+
+  // Getting a user's notifications using their id
+  public static Collection<Notification> getUserNotifications(String id) {
+    Collection<Notification> notifications = new ArrayList<>();
+
+    Filter idFilter = new FilterPredicate("userId", FilterOperator.EQUAL, id);
+
+    // Querying all of the user's notifications from the Datastore.
+    Query query = new Query("Notification").setFilter(idFilter)
+      .addSort("timestamp",SortDirection.DESCENDING);
+
+    PreparedQuery results = datastore.prepare(query);
+
+    // Populating the list
+    for (Entity entity : results.asIterable()) {
+      String firstId = (String) entity.getProperty("userId");
+      String secondId = (String) entity.getProperty("otherUserId");
+      long timestamp = (long) entity.getProperty("timestamp");
+      String notificationType = (String) entity.getProperty("type");
+       
+      Notification notification;
+
+      if (notificationType.equals("matching")) {
+        notification = new MatchNotification(firstId, secondId, timestamp);
+      } else {
+        notification = new MessageNotification(firstId, secondId, timestamp);
+      }
```

Comment:
I think there should be a if/if else/else here in case the notificationType isn't "matching" or "message". The new else can just say something like "There is an inconsistency in the database"